### ISSUE TITLE
Name 'test' is not defined

### ISSUE THREAD

**Issue by rdtm443** [emirhanai/Cryptocurrency-Prediction-with-Artificial-Intelligence#1]
Hey there

Many thanks for this project - trying to work on it, but found that 'test' is not defined, please see screenshot:

![image](https://user-images.githubusercontent.com/25002690/163460465-9d4d31de-2850-4c6a-9aa3-d45c11bdc864.png)

What dataset are you referring to? 'test_data'? 

Thanks a lot!

Edit: oh, and by the way, how can I really get the prediction graph to be ahead like on your sample? For now the it is lagging behind the real curve... (after changing 'test' to 'test_data')

**Comment by hedijezi** [comment]
hello 

after running this cell: 
model = build_lstm_model(
    X_train, output_size=1, neurons=lstm_neurons, dropout=dropout, loss=loss,
    optimizer=optimizer)
modelfit = model.fit(
    X_train, y_train, validation_data=(X_test, y_test), epochs=epochs, batch_size=batch_size, verbose=1, shuffle=False)
   
i got 
Epoch 1/20
266/266 [==============================] - 6s 10ms/step - loss: nan - val_loss: nan
Epoch 2/20
266/266 [==============================] - 2s 9ms/step - loss: nan - val_loss: nan
Epoch 3/20
266/266 [==============================] - 2s 8ms/step - loss: nan - val_loss: nan
Epoch 4/20
266/266 [==============================] - 2s 9ms/step - loss: nan - val_loss: nan
Epoch 5/20
266/266 [==============================] - 2s 8ms/step - loss: nan - val_loss: nan
Epoch 6/20
266/266 [==============================] - 2s 9ms/step - loss: nan - val_loss: nan
Epoch 7/20
266/266 [==============================] - 2s 7ms/step - loss: nan - val_loss: nan
Epoch 8/20
266/266 [==============================] - 2s 8ms/step - loss: nan - val_loss: nan
Epoch 9/20
266/266 [==============================] - 3s 9ms/step - loss: nan - val_loss: nan
Epoch 10/20
266/266 [==============================] - 2s 9ms/step - loss: nan - val_loss: nan
Epoch 11/20
266/266 [==============================] - 2s 9ms/step - loss: nan - val_loss: nan
Epoch 12/20
266/266 [==============================] - 2s 8ms/step - loss: nan - val_loss: na ( etc ............ )

u had the same problem ? if not do u know what could be the problem ? thanks in advance !

**Comment by emirhanai** [comment]
Hey @hedijezi , you should use Normalize function from sklearn or you should write normalize algorithm, okey? If you do it, problem will solve.
Hey @rdtm443 thank you for your feedback. After then, you can change normalize algorithm or you can change data type (open-close-high-low etc.)

**Comment by emirhanai** [comment]
Problem solved.
